src/fdb5/tools/fdb-root.cc

```diff
@@ -65,16 +65,16 @@ void FdbRoot::execute(const eckit::option::CmdArgs& args) {
 
             Config conf = config(args);
             const Schema& schema = conf.schema();
-            Key result;
+            TypedKey result{conf.schema().registry()};
```

Comment:
dbReader and dbWriter (below) could share a `const auto key = result.canonical();`